src/controllers/userController.js

```diff
@@ -309,5 +344,135 @@ export const logout = async (req, res) => {
       });
       res.status(200).json({ message: 'Logout successful' });
     }
-  };
+  }
+};
+
+export const enable2FA = async (req, res) => {
+  try {
+    const user = await User.findById(req.userId);
+
+    const secret = speakeasy.generateSecret({ name: 'CloudMedix' });
+
+    user.totpSecret = secret.base32;
+    await user.save();
+
+    qrcode.toDataURL(secret.otpauth_url, (err, qrCodeUrl) => {
+      if (err) {
+        logger.error('Error generating QR code', {
+          method: req.method,
+          url: req.originalUrl,
+          error: err,
+          userId: req.userId,
+        });
+        return res.status(500).json({ message: 'Internal server error' });
+      }
+
+      return res
+        .status(200)
+        .json({ message: '2FA enabled successfully', qrCodeUrl });
+    });
+  } catch (error) {
+    logger.error('Error enabling 2FA', {
+      method: req.method,
+      url: req.originalUrl,
+      error: error.message,
+      userId: req.userId,
+    });
+    res.status(500).json({ message: 'Internal server error' });
+  }
+};
+
+export const verify2FA = async (req, res) => {
+  const { userId, totpToken } = req.body;
+
+  try {
+    const sessionKey = `2fa_pending:${userId}:${req.ip}`;
+
+    const sessionExists = await redisClient.exists(sessionKey);
+    if (!sessionExists) {
+      return res
+        .status(403)
+        .json({ message: '2FA session expired or invalid' });
+    }
```

Comment:
should this be a 403 or a 400? Im not sure. Technically the user has acces to this endpoint. The same goes for "2FA not enabled for this user' ". I think this second one should be a 403